### DISCUSSION TITLE
Disable serving the default metrics endpoint when using a dedicated metrics_server

### DISCUSSION THREAD

**Discussion by holgerw** [akoutmos/prom_ex#76] [Q&A] [answered]
Should it be possible to disable serving the default metrics endpoint (ex: localhost:4000/metrics), when we configure prom_ex with a dedicated metrics_server (ex: metrics_server: [port: 4021])?

Right now metrics are served at both endpoints "localhost:4000/metrics" and "localhost:4021/metrics".

As the main motivation for usage of a dedicated metrics_server is to protect access at the network layer (firewall) when we use a local prometheus scrapper like the Grafana Agent, it would be great if serving metrics at the main endpoint could be disabled.

**Comment by akoutmos** [comment, marked as answer]
Hey there! PromEx will only expose metrics when configured to do so (via the provided plug or the standalone server). My guess is that you still have your plug entry in `endpoint.ex`? Kinda like https://github.com/akoutmos/prom_ex/blob/master/example_applications/web_app/lib/web_app_web/endpoint.ex#L29

**Reply by holgerw** [reply]
That solved it. Thank you for your work on prom_ex. Super helpful to get comprehensive monitoring up and running rapidly. I like especially that prom_ex creates a Grafana annotation on application start, with the GIT_SHA, if configured. Makes it possible to trace performance regressions to deployments and code versions.

**Reply by akoutmos** [reply]
Glad that you are getting some good use out of PromEx! . This is very much a culmination of a few years of personal pain points all being addressed in 1 library lol.

If you are or the company you work for are interested in supporting the project be sure to check out the [Sponsorship](https://github.com/sponsors/akoutmos) page :).

**Reply by holgerw** [reply]
I imagine the effort that has gone into the library. Just on the Grafana side, constructing the parameterized dashboards, and annotating the graphs, requires digging through much documentation. Grafana's flexibility has its toll. And here I see much of the value of prom_ex. I did not dive into the Elixir and Telemetry part. But imagine that it also includes many lessons learned from problems along the way.
Currently I am setting prom_ex up in the context of a side project. If it catches up, at least with usage or even with revenue, for sure I want to give back. I have to see.
I also imagine that using promex and the Grafana dashboards regularly feature ideas will come up and I can contribute with PRs.

**Reply by akoutmos** [reply]
Best of luck with the side project and I appreciate any contributions that can be made back :)
